OpenRA.Mods.Common/Lint/CheckPlayers.cs

```diff
@@ -22,22 +22,26 @@ public void Run(Action<string> emitError, Action<string> emitWarning, Map map)
 			var players = new MapPlayers(map.PlayerDefinitions).Players;
 
 			var playerNames = players.Values.Select(p => p.Name);
-			foreach (var player in players)
-				foreach (var ally in player.Value.Allies)
+			foreach (var player in players.Values)
+			{
+				foreach (var ally in player.Allies)
 					if (!playerNames.Contains(ally))
 						emitError("Allies contains player {0} that is not in list.".F(ally));
 
-			foreach (var player in players)
-				foreach (var enemy in player.Value.Enemies)
+				foreach (var enemy in player.Enemies)
 					if (!playerNames.Contains(enemy))
 						emitError("Enemies contains player {0} that is not in list.".F(enemy));
 
+				if (player.OwnsWorld && (player.Enemies.Any() || player.Allies.Any()))
+					emitWarning("The player {0} owning the world should not have any allies or enemies.".F(player.Name));
+			}
+
 			var worldActor = map.Rules.Actors["world"];
 
 			var races = worldActor.Traits.WithInterface<CountryInfo>().Select(c => c.Race);
-			foreach (var player in players)
-				if (!string.IsNullOrWhiteSpace(player.Value.Race) && player.Value.Race != "Random" && !races.Contains(player.Value.Race))
-					emitError("Invalid race {0} chosen for player {1}.".F(player.Value.Race, player.Value.Name));
+			foreach (var player in players.Values)
+				if (!string.IsNullOrWhiteSpace(player.Race) && player.Race != "Random" && !races.Contains(player.Race))
```

Comment:
The hard-coded check for the random race isn't needed anymore. "Random" is a legit race in its own right now.